File: obelisk_ws/src/cpp/obelisk_leap_cpp/src/leap_example_controller.cpp
#include "leap_example_position_setpoint_controller.h"
#include "obelisk_ros_utils.h"

int main(int argc, char* argv[]) {
    obelisk::utils::SpinObelisk<obelisk::LeapExamplePositionSetpointController,
                                rclcpp::executors::SingleThreadedExecutor>(argc, argv,
                                                                           "leap_position_setpoint_controller");
}

File: obelisk_ws/src/python/hardware/obelisk_leap_py/obelisk_leap_py/leap_example_controller.py
from typing import List, Optional

from rclpy.executors import SingleThreadedExecutor

from obelisk_py.core.utils.ros import spin_obelisk
from obelisk_py.zoo.control.example.leap_example_pos_setpoint_controller import LeapExamplePositionSetpointController


def main(args: Optional[List] = None) -> None:
    """Main entrypoint."""
    spin_obelisk(args, LeapExamplePositionSetpointController, SingleThreadedExecutor)


if __name__ == "__main__":
    main()
